queue.go

```diff
@@ -226,6 +231,37 @@ func (queue *Queue) Purge() {
 			}
 		}
 	}()
+
+	return &waitGroup
+}
+
+// Goes beyond `Purge` behaviour to synchronously delete all tasks and their name handles
+func (queue *Queue) HardReset(s *Server) {
+	waitGroup := queue.Purge()
+	waitGroup.Wait()
+
+	// This is still a bit awkward - we can't *guarantee* the task is fully deleted even after the WaitGroup because:
+	// - Purge() calls task.Delete()
+	// - task.Delete() writes to a buffered `cancel` channel
+	// - task.Schedule() reads from that buffered channel in a separate goroutine
+	// - When that goroutine sees the task is cancelled, it sets the task value to nil in the tasks map
+	//
+	// We need to be certain that we only remove the task from map *after* that completes, otherwise the task name will
+	// be reinserted with the nil value. At the moment the only easy way I can think of is to sleep for a very short
+	// period to allow the tasks' internal goroutines to fire first.
+	time.Sleep(10 * time.Millisecond)
```

Comment:
Making sure that the task is removed from the list *after* the task.onDone callback fires to set it to nil caused me a bit of trouble - I couldn't see a reliable way to actually wait for that to happen without quite a bit of refactoring (but my go concurrency knowledge is very limited, so I might well have missed something).

Adding a very short sleep here seems to be working reliably - I've run it a lot of times and never seen a race condition happen.